src/Microsoft.AspNet.Razor.Runtime/XMLDocumentationProvider.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if !DNXCORE50 // Not used in coreclr
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Xml.Linq;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Razor.Runtime
+{
+    public class XmlDocumentationProvider
+    {
+        private readonly IEnumerable<XElement> _members;
+
+        public XmlDocumentationProvider(string xmlFileLocation)
+        {
+            // XML file processing is defined by: https://msdn.microsoft.com/en-us/library/fsbx0t7x.aspx
+            var xmlDocumentation = XDocument.Load(xmlFileLocation);
+            var documentationRootMembers = xmlDocumentation.Root.Element("members");
+            _members = documentationRootMembers.Elements("member");
+        }
+
+        public string GetSummary(string id)
+        {
+            var associatedMemeber = GetMember(id);
+
+            return associatedMemeber?.Element("summary")?.Value.Trim();
+        }
+
+        public string GetRemarks(string id)
+        {
+            var associatedMemeber = GetMember(id);
+
+            return associatedMemeber?.Element("remarks")?.Value.Trim();
+        }
+
+        public bool HasDocumentation(string id)
+        {
+            return GetMember(id) != null;
```

Comment:
`<member>` may not contain `<summary>` or `<remarks>`. Either check that one or both exist or change method name to something like `MayHaveDocumentation()`. But best option is probably to remove this method and change caller to confirm at least one of `summary` or `remarks` are not `null` or empty before creating a usage descriptor.